File: ruby/byte/binarygap.rb
String myint = 9.to_s(2)
p "Binary: #{myint}"
Integer i = myint.size
Integer totalbinarygap = 0
Integer startcount = 0
loop do
    if i >= 0            
        case myint[i]
        when "0"
            if startcount == 1            
                totalbinarygap = totalbinarygap + 1
                startcount = 0
            end
        when "1"
            startcount = 1
        else
            totalbinarygap = totalbinarygap
        end
    else
        break   
    end
    i = i - 1
end
p "Total binarygap: #{totalbinarygap}" 

File: java/BinaryGap.java
public class BinaryGap 
{
    public static void main( String[] args )
    {
        int myint = 9654;
        String mybyte = Integer.toBinaryString(myint);
        System.out.println("Binary: " + mybyte);
        int totalbinarygap = 0;        
        boolean binarystartcount = false;
        for (int i = (mybyte.length()-1); i >= 0; i--) {            
            switch (String.valueOf(mybyte.charAt(i))) {
                case "0":
                    if (binarystartcount) {
                        totalbinarygap = totalbinarygap + 1;
                        binarystartcount = false;
                    }                    
                    break;
                case "1":                    
                    binarystartcount = true;
                    break;
                default:
                    break;
            }
        }
        System.out.println("Binary Gap: " +totalbinarygap);        
    }
}